server/src/main/java/org/apache/druid/server/scheduling/ThresholdBasedQueryPrioritizationStrategy.java

```diff
@@ -51,13 +53,15 @@ public class ThresholdBasedQueryPrioritizationStrategy implements QueryPrioritiz
   private final Optional<Duration> periodThreshold;
   private final Optional<Duration> durationThreshold;
   private final Optional<Duration> segmentRangeThreshold;
+  private final Set<String> exemptDatasources;
 
   @JsonCreator
   public ThresholdBasedQueryPrioritizationStrategy(
       @JsonProperty("periodThreshold") @Nullable String periodThresholdString,
       @JsonProperty("durationThreshold") @Nullable String durationThresholdString,
       @JsonProperty("segmentCountThreshold") @Nullable Integer segmentCountThreshold,
       @JsonProperty("segmentRangeThreshold") @Nullable String segmentRangeThresholdString,
+      @JsonProperty("exemptDatasources") @Nullable Set<String> exemptDatasources,
```

Comment:
This might lead to more issues like one query hogging the cluster for the exempt data source and all your query lanning goes for a toss. Should we revert the logic, ie if an "lowPriorityDataSource" is hit, lower the priority always. That way the cluster can be saved against outages ? Its better to err on the side of caution no ?

Since there is not clear answer here : 
* What do other data bases do ?
* What's your practical experience running this config in production.